### PR TITLE
Bitcoin Clashic ABCD 0.16.1

### PR DESCRIPTION
**Pull request by clashicly**
Started branch as 0.15.101, but on User Agent it came up at 0.16.1.
Fixed the issue with versioning.
So it shall be Bitcoin Clashic ABCD 0.16.1

This is the sum total of Rebranding. This client is designed to run side by side with Bitcoin Core, Bitcoin ABC, Litecoin core, etc.

- Symbol change from BCC to BCL
- New splash page and icons (if you don't like the look, please contribute!)
- Rename all files from bitcoin to clashic. ie. bitcoin-qt -> clashic-qt
- Rename manpages
- Update default data directory from .bitcoin -> .clashic
- Add 2000 to all port numbers
- Windows registry changes
- Added seeder node seeder.clashic.cash

**It does not contain consensus changes. No hardfork in this pull request.**

### CODE REVIEW

**Comment by clashicly** [comment]
It seems clashic-qt is able to connect to nodes running on port 8333
But clashicd seems to only try to connect to port 10333 when retrieving nodes from Bitcoin Cash seeders

`# tail -f debug.log
2018-04-30 17:37:44 connect() to 128.199.164.172:10333 failed after select(): Connection refused (111)
2018-04-30 17:37:51 connect() to 46.4.33.85:10333 failed after select(): Connection refused (111)
2018-04-30 17:37:57 connect() to 192.203.228.91:10333 failed after select(): Connection refused (111)
2018-04-30 17:37:57 connect() to 45.63.39.112:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:03 connect() to 195.154.177.49:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:10 connect() to 47.95.32.168:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:10 connect() to 47.94.47.209:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:22 connect() to 47.94.45.184:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:22 connect() to 192.187.121.43:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:28 connect() to 199.101.100.58:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:40 connect() to 54.36.176.224:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:41 connect() to 172.104.151.66:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:41 connect() to 192.99.19.160:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:42 connect() to 45.77.43.164:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:43 connect() to 67.221.142.252:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:43 connect() to 206.51.235.198:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:44 connect() to 47.88.57.99:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:44 connect() to 138.197.160.144:10333 failed after select(): Connection refused (111)
2018-04-30 17:38:50 connect() to 109.233.58.46:10333 failed after select(): No route to host (113)
`

Therefore it gets no blocks
`# clashic-cli -rpcconnect=127.0.0.1 -rpcport=10332 -rpcuser=somerpcuser -rpcpassword=somerpcpassword getblockchaininfo
{
  "chain": "main",
  "blocks": 0,
  "headers": 0,
  "bestblockhash": "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f",
  "difficulty": 1,
  "mediantime": 1231006505,
  "verificationprogress": 3.147261566546461e-09,
  "chainwork": "0000000000000000000000000000000000000000000000000000000100010001",
  "pruned": true,
  "softforks": [
    {
      "id": "bip34",
      "version": 2,
      "reject": {
        "status": false
      }
    },
    {
      "id": "bip66",
      "version": 3,
      "reject": {
        "status": false
      }
    },
    {
      "id": "bip65",
      "version": 4,
      "reject": {
        "status": false
      }
    }
  ],
  "bip9_softforks": {
    "csv": {
      "status": "defined",
      "startTime": 1462060800,
      "timeout": 1493596800,
      "since": 0
    }
  },
  "pruneheight": 0
}`

only way to jumpstart is explicitly adding via clashic-cli
`# clashic-cli -rpcconnect=127.0.0.1 -rpcport=10332 -rpcuser=somerpcuser -rpcpassword=somerpcpassword addnode xxx.xxx.xxx.xxx:8333 add
`

Then the debug.log starts flooding in
`2018-04-30 18:00:08 UpdateTip: new best=000000001786abd75dc912d8eabe85080c7e822858d445644fa3a3e059c2033b height=51729 version=0x00000001 log2_work=48.571264 tx=53868 date='2010-04-18 18:14:28' progress=0.000170 cache=10.8MiB(53381txo)
2018-04-30 18:00:08 UpdateTip: new best=000000001e4405ae38eeedceafcced6e8108d2c63f07999b26026d2e16e9f5b2 height=51730 version=0x00000001 log2_work=48.57138 tx=53982 date='2010-04-18 18:19:38' progress=0.000170 cache=10.8MiB(53494txo)
2018-04-30 18:00:08 UpdateTip: new best=00000000071ca88b41fe390be49c489b325aab404605bca51c0c499ad824cdfb height=51731 version=0x00000001 log2_work=48.571496 tx=54070 date='2010-04-18 18:28:28' progress=0.000170 cache=10.8MiB(53582txo)
2018-04-30 18:00:08 UpdateTip: new best=0000000008c5ebcd04a8303c6c35c84e0560ef5a3e1ca34c59e356216eecc4c8 height=51732 version=0x00000001 log2_work=48.571612 tx=54073 date='2010-04-18 18:29:17' progress=0.000170 cache=10.8MiB(53585txo)
2018-04-30 18:00:08 UpdateTip: new best=000000001df6a4e1a1634ac4bf11a80a7444ee1f0b46543c22a827fdfb12b559 height=51733 version=0x00000001 log2_work=48.571728 tx=54092 date='2010-04-18 18:29:16' progress=0.000170 cache=10.8MiB(53604txo)
2018-04-30 18:00:08 UpdateTip: new best=00000000042a3aee0691a8705cf96536e8ad71b01a3deddb82e1ebc1c4540053 height=51734 version=0x00000001 log2_work=48.571843 tx=54491 date='2010-04-18 18:32:21' progress=0.000171 cache=10.9MiB(54002txo)
2018-04-30 18:00:08 UpdateTip: new best=000000001314b2b56784f69bedc782a858bff9909f495a3c06066f865b092f1a height=51735 version=0x00000001 log2_work=48.571959 tx=54654 date='2010-04-18 18:36:33' progress=0.000172 cache=10.9MiB(54165txo)
2018-04-30 18:00:08 UpdateTip: new best=0000000006cb0e1cdd14238e8b55140c0e743ea11e9479d029e48bb63175ce30 height=51736 version=0x00000001 log2_work=48.572075 tx=54674 date='2010-04-18 18:38:04' progress=0.000172 cache=10.9MiB(54185txo)
`

**Comment by clashicly** [comment]
testnet also attempts to use seeder supplied addresses, but with port 20333

`2018-05-06 11:52:24 opencon thread start
2018-05-06 11:52:24 init message: Done loading
2018-05-06 11:52:24 msghand thread start
2018-05-06 11:52:51 19 addresses found from DNS seeds
2018-05-06 11:52:51 dnsseed thread exit
2018-05-06 11:52:56 connect() to 52.7.233.13:20333 failed after select(): Connection refused (111)
2018-05-06 11:53:13 connect() to 188.40.93.205:20333 failed after select(): Connection refused (111)
2018-05-06 11:53:14 connect() to 37.233.101.121:20333 failed after select(): Connection refused (111)
2018-05-06 11:53:53 connect() to 198.245.51.212:20333 failed after select(): Connection refused (111)
2018-05-06 11:53:59 connect() to 47.91.198.174:20333 failed after select(): Connection refused (111)
2018-05-06 11:54:11 connect() to 37.233.101.121:20333 failed after select(): Connection refused (111)`

File: README.md [hunk 1]
```diff
@@ -11,28 +11,27 @@ anyone, anywhere in the world. Bitcoin uses peer-to-peer technology to operate
...
-What is Bitcoin ABC?
---------------------
+What is Bitcoin Clashic ABCD?
+-----------------------------
 
-Bitcoin ABC is the name of open source software which enables the use of Bitcoin.
+Bitcoin Clashic ABCD is the name of open source software which enables the use of Bitcoin.
 It is designed to facilite a hard fork to increase Bitcoin's block size limit.
-"ABC" stands for "Adjustable Blocksize Cap".
+"ABCD" stands for "Always Be Clashic D......".
```
Comment (cryptomeow) [inline README.md:19]: Dear? :stuck_out_tongue:

File: configure.ac [hunk 2]
```diff
@@ -2,23 +2,23 @@ dnl require autoconf 2.60 (AS_ECHO/AS_ECHO_N)
 AC_PREREQ([2.60])
 define(_CLIENT_VERSION_MAJOR, 0)
 define(_CLIENT_VERSION_MINOR, 15)
```
Comment (cryptomeow) [inline configure.ac]: 16?

File: configure.ac [hunk 3]
```diff
@@ -2,23 +2,23 @@ dnl require autoconf 2.60 (AS_ECHO/AS_ECHO_N)
...
-define(_CLIENT_VERSION_REVISION, 1)
+define(_CLIENT_VERSION_REVISION, 101)
```
Comment (cryptomeow) [inline configure.ac]: 1?

File: doc/Doxyfile [hunk 4]
```diff
@@ -32,19 +32,19 @@ DOXYFILE_ENCODING      = UTF-8
...
 
-PROJECT_NAME           = "Bitcoin ABC"
+PROJECT_NAME           = "Bitcoin Clashic ABCD"
 
 # The PROJECT_NUMBER tag can be used to enter a project or revision number. This
 # could be handy for archiving the generated documentation or if some version
 # control system is used.
 
-PROJECT_NUMBER         = 0.15.1
+PROJECT_NUMBER         = 0.15.101
```
Comment (cryptomeow) [inline doc/Doxyfile]: 0.16.1 ?

File: doc/release-notes.md [hunk 5]
```diff
@@ -1,18 +1,3 @@
-Bitcoin ABC version 0.15.1 is now available from:
+Bitcoin Clashic ABCD version 0.15.101 is now available from:
```
Comment (cryptomeow) [inline doc/release-notes.md]: 0.16.1 ?

File: doc/release-notes.md [hunk 6]
```diff
@@ -1,18 +1,3 @@
...
-- Display if a node is using Cash magic in getpeerinfo (D546).
-- Use Cash magic by default when establishing connections (D547).
-- Add seeder to Bitcoin ABC repository. From Bitcoin Seeder by Pieter Wiulle.
-  (D559, D560, D561, D562, D564, D565, D566, D568, D579, D585)
-- Ensure backupwallet fails if target is the same as source (D550).
-- Upgrade to LevelDB 1.20. Port of Core 10544 and 10958 (D580, D584).
-- Various backports from Core.
-- Various bug fixes.
-- Various style fixes, code cleanups, and refactorings.
+  <https://bitcoin.clashic.cash/downloads/0.15.101/>
```
Comment (cryptomeow) [inline doc/release-notes.md]: We can just use Github's releases existing infra (see tag releases & release assets) and/or upload them on https://github.com/Bitcoin-Clashic/bitcoin-clashic.github.io

File: src/chainparams.cpp [hunk 7]
```diff
@@ -156,7 +159,7 @@ class CMainParams : public CChainParams {
         pchCashMessageStart[1] = 0xe1;
         pchCashMessageStart[2] = 0xf3;
         pchCashMessageStart[3] = 0xe8;
-        nDefaultPort = 8333;
+        nDefaultPort = 10333;
```
Comment (cryptomeow) [inline src/chainparams.cpp:162]: :+1:

File: src/chainparams.cpp [hunk 8]
```diff
@@ -320,7 +323,7 @@ class CTestNetParams : public CChainParams {
         pchCashMessageStart[1] = 0xe5;
         pchCashMessageStart[2] = 0xf3;
         pchCashMessageStart[3] = 0xf4;
-        nDefaultPort = 18333;
+        nDefaultPort = 20333;
```
Comment (cryptomeow) [inline src/chainparams.cpp:326]: :+1:

File: src/chainparams.cpp [hunk 9]
```diff
@@ -439,7 +442,7 @@ class CRegTestParams : public CChainParams {
         pchCashMessageStart[1] = 0xb5;
         pchCashMessageStart[2] = 0xbf;
         pchCashMessageStart[3] = 0xfa;
-        nDefaultPort = 18444;
+        nDefaultPort = 20444;
```
Comment (cryptomeow) [inline src/chainparams.cpp:445]: :+1:

File: src/clientversion.h [hunk 10]
```diff
@@ -13,11 +13,11 @@
...
  */
 
-//! These need to be macros, as clientversion.cpp's and bitcoin*-res.rc's voodoo
+//! These need to be macros, as clientversion.cpp's and clashic*-res.rc's voodoo
 //! requires it
 #define CLIENT_VERSION_MAJOR 0
-#define CLIENT_VERSION_MINOR 15
-#define CLIENT_VERSION_REVISION 1
+#define CLIENT_VERSION_MINOR 16
+#define CLIENT_VERSION_REVISION 0
```
Comment (cryptomeow) [inline src/clientversion.h]: 1?

File: src/consensus/params.h [hunk 11]
```diff
@@ -51,6 +51,8 @@ struct Params {
     int uahfHeight;
     /** Block height at which OP_RETURN replay protection stops */
     int antiReplayOpReturnSunsetHeight;
+    /** Block height at which TVS43 becomes active */
+    int TVS43Height;
```
Comment (cryptomeow) [inline src/consensus/params.h:55]: TVS43 ?
Maybe a tiny addition on https://github.com/Bitcoin-Clashic/tvs ?

**Review by cryptomeow — COMMENTED** [review]
Amazing work @clashicly :bowing_man: 
Truly keeping it clashic ;) :+1: 

A few nits/comments.
utACK c03536f

I'll build on a linux machine soon.

File: src/consensus/params.h (hunk 11 above)
Comment (clashicly) [reply src/consensus/params.h:55]: Work in progress..
https://github.com/clashicly/tvs/blob/patch-1/tvs-0043.mediawiki

**Comment by cryptomeow** [comment]
@clashicly can you make another PR against `master`? I reset it and also added travis integration to the repo.

**Pull request merged**